### ISSUE TITLE
OpenID Connect plugin - high level design

### ISSUE THREAD

**Issue by altexy** [GluuFederation/gluu-gateway#283]
```
Michael Schwartz nynymike admin gluu 2:21 PM
The key here is how to use both OpenID Connect and UMA together.
If we are just building an OpenID Connect proxy, there is no point: the existing NGINX plugin does the job fine.
For me, the key is how can we use UMA access tokens to protect web pages.
But before we can do this, we need the authentication phase to know who is this person.
Using UMA tokens, we can define policies centrally about who can access which web pages.
This would be an incredibly powerful feature.
Also, oxd is a good openid connect client


Michael Schwartz nynymike admin gluu 2:26 PM
it’s not as hard as it sounds.
We can use the Authorization header
OpenID Connect does not need this anyway
It can carry the UMA RPT access token
the OpenID Connect access token is used by oxd
it is not needed for HTTP requests to the web proxy
but yes, let’s think it through
the goal is not OpenID Connect … its OpenID + UMA
```

**Comment by altexy** [comment]
@nynymike 
Should we support only authorization code flow?

Can you provide an example diagram or description of using OpenID Connect and UMA together?

Using authorization code flow user may be authenticated, but I don't have any idea how a human obtains UMA RPT token.

**Comment by altexy** [comment]
https://github.com/GluuFederation/gluu-gateway/blob/master/kong-openid-rp/doc/konga-openid-wireframe.png

**Comment by altexy** [comment]
@ldeveloperl1985 should we close this?

**Comment by ldeveloperl1985** [comment]
Yes
